lib/WebpackOptionsApply.js

```diff
@@ -425,9 +425,14 @@ class WebpackOptionsApply extends OptionsApply {
 				options.optimization.sideEffects === true
 			).apply(compiler);
 		}
-		if (options.optimization.providedExports) {
+		if (
+			options.optimization.providedExports ||
+			(options.optimization.providedExports === false &&
+				options.output.library &&
+				options.output.library.type === "module")
```

Comment:
I dont' think we should rely on such things, we need to improve logic inside the plugin itself

Reply:
Yes, i also think this would destroy the meaning of the config, but 'output: module' does rely on the module's exportsInfo, maybe we need to simply record export by sth like 'exportsInfo' for 'output: module'.

Reply:
> Yes, i think this would also destroy the meaning of the config, but 'output: module' does rely on the module's exportsInfo, maybe we need to simply record export by sth like 'exportsInfo' for 'output: module'.

And it seems that users rarely set 'providedExport' to false because of the performance issues of it.